EIPS/eip-7928.md

```diff
@@ -0,0 +1,199 @@
+
+Client implementations MUST compare the accessed addresses and storage keys gathered during execution (as defined in [EIP-2929](./eip-2929.md)) with those included in the BAL to determine validity.
+
+Client implementations MAY invalidate the block right away in case any transaction steps outside the declared state.
+
+## Rationale
+
+### BAL Design Choice
+
+We chose the BAL variant for several key reasons:
+
+1. **Balance between size and parallelization benefits**: BALs enable both parallel disk reads and parallel EVM execution while keeping the size manageable.
+
+2. **Storage value inclusion**: Including post-execution values for write operations facilitates state reconstruction during syncing, enabling faster chain catch-up.
+
+3. **Balance diffs**: Including balance diffs ensures correct handling of execution that depends on account balances, crucial for parallel transaction execution.
+
+4. **Nonce diffs**: Providing pre-block nonce values for transaction senders enables validators to verify nonce ordering without sequential execution.
+
+5. **Efficiency**: An analysis of historical blocks (random sample of 1000 blocks between 22,195,599 and 22,236,441) shows that BALs would have had an average size of around 57 KiB with balance diffs adding only 9.6 KiB on average, which is a reasonable overhead given the performance benefits.
+
+6. **Transaction independence**: Analysis of the same block sample revealed that approximately 60-80% of transactions within a block are fully independent from one another, meaning they access disjoint sets of addresses and storage slots.
+
+### Block Size Considerations
+
+Including access lists increases block size, potentially impacting network propagation times. Based on analysis of historical blocks:
+
+- Average BAL size over 1,000 blocks was around 57 KiB (SSZ-encoded)
+- Average balance diff size was approximately 9.6 KiB
+- Worst-case BAL size for consuming the entire block gas limit with storage access operations would be approximately 0.93 MiB
+- Worst-case balance diff size would be around 0.11 MiB
+
+These sizes are manageable and less than the current worst-case block size achievable through calldata.
+
+### Asynchronous Validation
+
+Block execution can proceed with parallel IO and parallel EVM operations, with verification of the access list occurring alongside execution, ensuring correctness without delaying block processing.
+
+## Backwards Compatibility
+
+This proposal requires changes to the block structure that are not backwards compatible and require a hard fork.
+
+## Security Considerations
+
+### Validation Overhead
+
+Nodes MUST verify that the access lists and balance diffs in the block accurately reflect the actual accesses and balance changes during execution. Discrepancies MUST result in block invalidation.
+
+### Block Size
```

Comment:
This will definitely be the major point and there should be a very thorough analysis on what the worst case will be. If it is less than X KiB on average then this is very nice, but if the worst-case scenario produces a 100MiB block then this will never get accepted. (So analysis should be included, and should convince everyone that all cases are covered and thus the mentioned worst case scenario is thus really the worst case scenario)

Reply:
The analysis I did in the shared post were already realistic and conservative. As soon as we settle on a design, I will start more thorougly analysing it. 
Still, I'm very certain we will be able to keep worst-case sizes below 1 MiB and the goal should be to stay below the calldata max, not because that's a magical boundary but more because we won't create another source of size-related bugs.